adala/utils/llm_utils.py

```diff
@@ -116,6 +122,30 @@ def handle_llm_exception(
     return _log_llm_exception(e), usage
 
 
+def _ensure_messages_fit_in_context_window(
+    messages: List[Dict[str, str]], model: str
```

Comment:
merge in master and make sure `self.canonical_model_string` is being passed in here for model name

Reply:
done